Item(by='jcelerier', descendants=None, kids=[24653380], score=None, time=1601548333, title=None, item_type='comment', url=None, parent=24648830, text='&gt; At least for python, using is not more limited, but more complete: It allows to handle exceptions that are raised in the block and it also allows you to raise exceptions in the &quot;destructor&quot;.<p>the difference is that with Python, the programmer can forget to use &quot;using&quot;. \nIn C++, as soon as you instantiate your variable on the stack somewhere, you know that ~T() will happen unless you get a segfault...<p>&gt;  For example a db transaction wrapper might want to rollback instead of commit if we leave the block via an exception.<p>I&#x27;d say that a transaction wrapper would have an explicit commit method which applies the transaction and &quot;empties&quot; the transactoin object - destructor would always rollback if there is something that was not committed - and if commit was called then there is nothing to rollback anymore so ~T() does nothing.')